### ISSUE TITLE
Not compile with Clang 15: error invalid suffix on literal; C++11 requires a space between literal and identifier 

### ISSUE THREAD

**Issue by AngryPenguinPL** [lensfun/lensfun#1954]
When trying compile lensfun 0.3.3 with Clang 15.0.6 on OpenMandriva Cooker I see build error:

```
DEBUG util.py:443:  /builddir/build/BUILD/lensfun-0.3.3/libs/lensfun/cpuid.cpp:95:18: error: invalid suffix on literal; C++11 requires a space between literal and identifier [-Wreserved-user-defined-literal]
DEBUG util.py:443:          "push %%"R_BX"\n" \
DEBUG util.py:443:                   ^
DEBUG util.py:443:                    
DEBUG util.py:443:  /builddir/build/BUILD/lensfun-0.3.3/libs/lensfun/cpuid.cpp:97:17: error: invalid suffix on literal; C++11 requires a space between literal and identifier [-Wreserved-user-defined-literal]
DEBUG util.py:443:          "pop %%"R_BX"\n" \
DEBUG util.py:443:                  ^
DEBUG util.py:443:                   
DEBUG util.py:443:  2 errors generated.
```

Full build log: https://file-store.openmandriva.org/api/v1/file_stores/c9296ed4d37005ddb33f914e40e45ee4055e6870.log?show=true

Worth to add previous unstable 0.3.95 compiled fine.
As workaround I switched right now to GCC and all works fine, but we trying use Clang for all packages and it would be good to Lensfun to compile with Clang.

**Comment by jonas-ott** [comment]
cherry-picking https://github.com/lensfun/lensfun/commit/694542f350fc7c9ccafa2d9acb4d4e00e690cbdc onto [release_0.3.x](https://github.com/lensfun/lensfun/tree/release_0.3.x) should fix it.

**Comment by Sturmflut** [comment]
I've backported 694542f350fc7c9ccafa2d9acb4d4e00e690cbdc (cherry-picking wasn't possible due to merge conflicts) and checked it with clang 12.0.1/13.0.1/14.0.6/15.0.2 and GCC 9.5.0/10.4.0/11.3.0/12.2.0. Could you please try against the linked branch?

**Comment by AngryPenguinPL** [comment]
Yes, I can confirm this fixed compilation with Clang.
Thanks

**Comment by Sturmflut** [comment]
Merged and lined up for 0.3.4.
